lib/spack/spack/test/cmd/install.py

```diff
@@ -311,6 +311,26 @@ def test_install_from_file(spec, concretize, error_code, tmpdir):
     assert install.returncode == error_code
 
 
+@pytest.mark.usefixtures('noop_install', 'config')
+@pytest.mark.parametrize('spec', [
+    (Spec('mpi')),
+    (Spec('mpi')),
+    (Spec('boost')),
+    (Spec('boost'))
+])
+def test_install_clispec_as_yaml_path(spec, tmpdir):
+    spec.concretize()
+
+    specfile = tmpdir.join('spec.yaml')
+
+    with specfile.open('w') as f:
+        spec.to_yaml(f)
+
+    # Test that we can install passing absolute path to spec yaml
+    install(str(specfile), fail_on_error=False)
```

Comment:
In addition to testing the case where we provide filenames, there should be a case for mixed specs and filenames to ensure it fails -- you can mark it xfail, though probably better in this case to make sure it raises the exception you expect.